### ISSUE TITLE
Accessibility issues found in ChatKit

### ISSUE THREAD

**Issue by paulkirow** [openai/chatkit-js#114]
I just had a short session with a blind user who primarily uses JAWS as a screen-reader and I thought I'd share some of the accessibility issues we encountered.

- h5 and h6 are used for the "You said:" and the "The assistant said:" titles, which is great,  although they don't always align with the existing hierarchy of the page (i.e. if the only other heading outside of the iframe was h1, it would jump to h5 in ChatKit).  Being able to customize the specific heading levels would be nice.  Even if we could provide just the starting level as an option then each heading level inside chatkit could flow from there.
- The title of the current chat is not a header,  if this was a header it would help greatly with screen-reader navigation. Most screen-reader users jump to headers to start their navigation, so the first header being the h5 "You Said:" means they often miss the content that comes before that.  There is a workaround for this issue,  which involves me fetching the title from the API and rendering it in a visually hidden header that appears just outside the iframe
- The copy button on the assistant message has no label, so screen-reader users aren't able to interpret what this button does.
- The file upload button is unlabelled,  and the root file input (that is visually hidden) is accessible when using the screen reader.
- When you send a message the focus is moved off the send button and back onto the textarea,  which I can understand because the button changes to a disabled loading icon.  There is no announcement to indicate something is loading or that their message is sent and now appears above.  Our screenreader user had to just intuit that they should go back to the previous h5 to see their new message in the chat area.  Doing something with aria-live could be beneficial here.
- Lastly, the visually hidden labels don't appear to be exposed for localization.  Messages like "The assistant said:" and "You said:" don't appear to translate,  even when providing the `locale` option to `useChatKit`

In our session we only evaluated the chat component of ChatKit and didn't look at the history page.

**Comment by jiwon-oai** [comment]
Thank you for the detailed descriptions of the accessibility issues! We'll work on fixing these.

**Comment by levyben** [comment]
The current accessibility hurdles are frustrating. What is the ETA for a fix?
